src/core/bpf-firewall.c

```diff
@@ -651,31 +651,24 @@ int bpf_firewall_load_custom(Unit *u) {
 }
 
 static int attach_custom_bpf_progs(Unit *u, const char *path, int attach_type, Set **set, Set **set_installed) {
-        BPFProgram *prog;
+        _cleanup_(bpf_program_freep) BPFProgram *prog;
         int r;
 
         assert(u);
 
-        set_clear(*set_installed);
+        set_clear_free(*set_installed);
+        *set_installed = set_new(&bpf_program_hash_ops);
```

Comment:
`set_clear_free` followed by `set_new` will result in an empty set: essentially same that is was before with `set_clear`.
`*set_installed` will store a different address though. What's the point of that?

Reply:
`bpf_program_hash_ops` is needed to specify the free function needed to free the `bpf_program` objects from the set. Previously, that was specified in `set_ensure_put`, but since here we instead use `SET_FOREACH_MOVE` we no longer can specify `bpf_program_hash_ops` except through `set_new`. It does appear that `set_clear` is sufficient though.